botocore/parsers.py

```diff
@@ -706,6 +706,7 @@ def _do_error_parse(self, response, shape):
         code = None
         if len(query_error_components) == 2 and query_error_components[0]:
             code = query_error_components[0]
+            error['Error']['Type'] = query_error_components[1]
```

Comment:
nit: do you think we should also declare the error structure at line#686 as
```
error = {"Error": {"Message": '', "Code": '', "Type": ''}, "ResponseMetadata": {}}
```

Reply:
No. In general for JSON protocols, errors returned do not have an equivalent `Type` field. So if the parsed error was initialized to always include a `Type` key, it will end up being an empty string for most services supported by the SDK.

Furthermore, it was a slight oversight that the parsed response for query protocol has an `Type` field in the first place (i.e. the SDK generally documents parsed error dictionaries has having a `Code` and `Message`). The shim of injecting the `Type` key is more for adding it back in case someone saw the field as part of the parsed response and added logic to utilize it. So I'm not sure if we want to expand its presence to other usages.

Reply:
Well explained. thanks. The PR LGTM. Please ask the repo owner to give you a go.